src/driver_arguments.cpp

```diff
@@ -117,21 +117,28 @@ std::string ConvArgsForMIOpenDriver(const miopen::TensorDescriptor& xDesc,
     std::stringstream ss;
     if(print_for_conv_driver)
         ConvDataType(ss, xDesc);
+
+    /// \todo Dimensions (N, C, H, W, K..) are always parsed as if layout is NC(D)HW.
+    /// For other layouts, invalid values are printed.
+
     if(convDesc.GetSpatialDimension() == 2)
     {
-        ss << " -n " << xDesc.GetLengths()[0] // clang-format off
-            << " -c " << xDesc.GetLengths()[1]
-            << " -H " << xDesc.GetLengths()[2]
-            << " -W " << xDesc.GetLengths()[3]
-            << " -k " << wDesc.GetLengths()[0]
-            << " -y " << wDesc.GetLengths()[2]
-            << " -x " << wDesc.GetLengths()[3]
-            << " -p " << convDesc.GetConvPads()[0]
-            << " -q " << convDesc.GetConvPads()[1]
-            << " -u " << convDesc.GetConvStrides()[0]
-            << " -v " << convDesc.GetConvStrides()[1]
-            << " -l " << convDesc.GetConvDilations()[0]
-            << " -j " << convDesc.GetConvDilations()[1]; // clang-format on
+        ss << " -n " << xDesc.GetLengths()[0] //
+           << " -c " << xDesc.GetLengths()[1] //
+           << " -H " << xDesc.GetLengths()[2] //
+           << " -W " << xDesc.GetLengths()[3] //
+           << " -k "
+           << (convDesc.mode == miopenTranspose        //
+                   ? wDesc.GetLengths()[1]             //
+                   : wDesc.GetLengths()[0])            //
```

Comment:
@JehandadKhan This resolves https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2459#issuecomment-1776230996